File: Client/Captions/CaptionsLib/source/tts/TTSWriter.vb
'Filename: TTSWriter.vb
'Version: 20121015

Imports CaptionsLib.Models
Imports CaptionsLib.TTS.TTSUtils

Imports System.IO

Namespace CaptionsLib.TTS

  Public Class TTSWriter
    Inherits BaseCaptionWriter

#Region "Methods"

    Public Overrides Sub WriteCaption(ByVal Caption As ICaption, ByVal writer As TextWriter)
      writer.WriteLine(CaptionToTTSString(Caption))
    End Sub

#End Region

  End Class

End Namespace

File: Client/Captions/CaptionsLib/source/tts/TTSWriter.cs
//Filename: TTSWriter.cs
//Version: 20131105

using System.IO;
using Microsoft.SilverlightMediaFramework.Core.Accessibility.Captions;

namespace ClipFlair.CaptionsLib.TTS
{

	public class TTSWriter : BaseCaptionWriter
	{

		#region --- Methods ---

		public override void WriteCaption(CaptionElement caption, TextWriter writer)
		{
			writer.WriteLine(TTSUtils.CaptionToTTSString(caption));
		}

		#endregion

	}

}